app/controllers/admin/users_controller.rb

```diff
@@ -46,8 +50,13 @@ def create
   def update
     respond_to do |format|
       if @user.update(user_params)
-        format.html { redirect_to admin_users_path, notice: 'User was successfully updated.' }
-        format.json { render :show, status: :ok, location: @user }
+        if current_user.role == 1
```

Comment:
Using `enumerate_it` you can use the easier to understand syntax:

`if current_user.role == UserRole::ADMIN`